lib/spatial_features/has_spatial_features/feature_import.rb

```diff
@@ -0,0 +1,50 @@
+require 'digest/md5'
+
+module SpatialFeatures
+  module FeatureImport
+    def update_features!(skip_invalid: false, **import_options)
+      ActiveRecord::Base.transaction do
+        imports = spatial_feature_imports(import_options)
+        cache_key = Digest::MD5.hexdigest(imports.collect(&:cache_key).join)
+        if has_spatial_features_hash? && cache_key != features_hash
+          import_features(imports)
+          validate_features!(skip_invalid)
+          update_attributes(:features_hash => cache_key)
+        elsif !has_spatial_features_hash?
+          import_features(imports)
+          validate_features!(skip_invalid)
+        end
+      end
+    end
+
+    private
+
+    def spatial_feature_imports(options = {})
+      spatial_features_options.fetch(:import, {}).collect do |data_method, importer_name|
+        data = send(data_method)
+        "SpatialFeatures::Importers::#{importer_name}".constantize.new(data, options) if data.present?
+      end.compact
+    end
+
+    def import_features(imports)
+      features.destroy_all
+      features << imports.flat_map(&:features)
+    end
+
+    def validate_features!(skip_invalid = false)
+      invalid = features.select {|feature| feature.errors.present? }
+      features.destroy(invalid)
```

Comment:
Add test for destroying invalid features. Does it explode when it sees a new_record?